### DISCUSSION TITLE
Clock signals inside chips

### DISCUSSION THREAD

**Discussion by AlphaCodingPilot** [UkrainianBanderasCat/Digital-Logic-Sim2#17] [Q&A] [answered]
Im trying to update clocks inside chips. If anyone knows chips are saved I whould like the custom chips the signals connected to a clock to update based on currentSignal field stored in simulation class whenever the ProcessOutput function Updates the chip.

**Comment by firecerne** [comment, marked as answer]
If we make clocks only work when powered they would work inside chips, and it would be really simple to implement

**Reply by AlphaCodingPilot** [reply]
I'm not sure that the clock whould change it's output inside an other chip though. It seems like a chip only uses AND and NOT gates to calculate it's output.

**Reply by firecerne** [reply]
It would, it works for every other chips so why not clocks ? It's an issue already present in the base game, for exemple if you put a NOT gate not connected to anything in it's input it won't work inside chips. I think the game only compute chips connected with an input.

**Reply by AlphaCodingPilot** [reply]
I tried it, it sadly doesn't work. Only AND and NOT gates work like that for some reason

**Reply by UkrainianBanderasCat** [reply]
But wait... in computer only one clock, and it connecting to chips which needs clock

**Reply by AlphaCodingPilot** [reply]
That's true. Maybe you could just not allow users to create a chip with a clock inside it.

**Reply by AlphaCodingPilot** [reply]
Or even better: it could replace the clock with an input pin when you create the chip.
